### DISCUSSION TITLE
Strange Behavior with Eloquent Query Refreshing

### DISCUSSION THREAD

**Discussion by matijarezek** [filamentphp/filament#10897] [Help] [answered]
Hey everyone,

I've encountered a peculiar issue with my Laravel application, and I'm hoping to get some insights. Here's a snippet of the code in question:

```
protected static ?string $model = Ticket::class;

public static function getEloquentQuery(): Builder
{
    if (Route::currentRouteName() == 'filament.admin.resources.tickets.index'){
        return static::getModel()::query()->where('is_archived', '0');
    }
    else{
        return static::getModel()::query()->where('is_archived', '1');
    }
}
```

The code is designed to return a specific Eloquent query based on the current route. If the route is 'filament.admin.resources.tickets.index', it retrieves the query with 'is_archived' set to '0', otherwise with 'is_archived' set to '1'. However, I'm facing an unexpected behavior: firstly I get the correcty query but after 5 seconds, the query seems to refresh, providing me with a different result.

I'd appreciate any help or suggestions on how to troubleshoot and fix this problem. Thanks in advance!

**Comment by leandrocfe** [comment, marked as answer]
Probably an issue with LW request. Why not filter tab?
https://filamentphp.com/docs/3.x/panels/resources/listing-records#using-tabs-to-filter-the-records

**Reply by matijarezek** [reply]
Yeah I guess that works, didn't even think of that. I was just complicating things.
Thank you so much!
